src/snowflake/snowpark/_internal/analyzer/schema_utils.py

```diff
@@ -2,21 +2,25 @@
 # Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
 #
 
-from typing import TYPE_CHECKING, List
+from typing import TYPE_CHECKING, List, Union
 
 import snowflake.snowpark
-from snowflake.connector.constants import FIELD_ID_TO_NAME
-from snowflake.connector.cursor import ResultMetadata
+from snowflake.connector.cursor import ResultMetadata, SnowflakeCursor
 from snowflake.snowpark._internal.analyzer.analyzer_utils import (
     quote_name_without_upper_casing,
 )
 from snowflake.snowpark._internal.analyzer.expression import Attribute
-from snowflake.snowpark._internal.type_utils import convert_sf_to_sp_type
+from snowflake.snowpark._internal.type_utils import convert_metadata_to_sp_type
 from snowflake.snowpark.types import DecimalType, LongType, StringType
 
 if TYPE_CHECKING:
     import snowflake.snowpark.session
 
+    try:
```

Comment:
Let's add a comment here describing the difference between `ResultMetadataV2` and `ResultMetadata` and noting that we intend to unify them in a future release.

Reply:
I have an explanation in the docstring for ResultMetadataV2, so I added a comment here pointing readers to that.